controllers/utils/utils.go

```diff
@@ -1,22 +1,66 @@
 package utils
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
 	"reflect"
 
 	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type IsvcDeploymentMode string
+
+var (
+	Serverless    IsvcDeploymentMode = "Serverless"
+	RawDeployment IsvcDeploymentMode = "RawDeployment"
+	ModelMesh     IsvcDeploymentMode = "ModelMesh"
 )
 
 const (
-	inferenceServiceDeploymentModeAnnotation      = "serving.kserve.io/deploymentMode"
-	inferenceServiceDeploymentModeAnnotationValue = "ModelMesh"
+	inferenceServiceDeploymentModeAnnotation = "serving.kserve.io/deploymentMode"
+	KserveConfigMapName                      = "inferenceservice-config"
 )
 
-func IsDeploymentModeForIsvcModelMesh(isvc *kservev1beta1.InferenceService) bool {
+func GetDeploymentModeForIsvc(ctx context.Context, cli client.Client, isvc *kservev1beta1.InferenceService) (IsvcDeploymentMode, error) {
+
+	// If ISVC specifically sets deployment mode using an annotation, return bool depending on value
 	value, exists := isvc.Annotations[inferenceServiceDeploymentModeAnnotation]
-	if exists && value == inferenceServiceDeploymentModeAnnotationValue {
-		return true
+	if exists {
+		switch value {
+		case string(ModelMesh):
+			return ModelMesh, nil
+		case string(Serverless):
+			return Serverless, nil
+		case string(RawDeployment):
+			return RawDeployment, nil
+		}
+	} else {
+		// ISVC does not specifically set deployment mode using an annotation, determine the default from configmap
+		controllerNs := os.Getenv("POD_NAMESPACE")
+		inferenceServiceConfigMap := &corev1.ConfigMap{}
+		err := cli.Get(ctx, client.ObjectKey{
+			Namespace: controllerNs,
+			Name:      KserveConfigMapName,
+		}, inferenceServiceConfigMap)
+		if err != nil {
+			return "", fmt.Errorf("error getting configmap 'inferenceservice-config'. %w", err)
+		}
+		var deployData map[string]interface{}
+		if err = json.Unmarshal([]byte(inferenceServiceConfigMap.Data["deploy"]), &deployData); err != nil {
+			return "", fmt.Errorf("error retrieving value for key 'deploy' from configmap %s. %w", KserveConfigMapName, err)
+		}
+		defaultDeploymentMode := deployData["defaultDeploymentMode"]
+		if defaultDeploymentMode == string(Serverless) {
+			return Serverless, nil
+		} else {
+			return RawDeployment, nil
+		}
 	}
-	return false
+	return "", nil
```

Comment:
Once you add a default to the switch up there, you can remove this line.

The callers of t his new function does not seem to take into account that an empty string would be returned. The behaviors seems to be undetermined in such case. So, let's not return an unexpected value.